astropy/units/structured.py

```diff
@@ -0,0 +1,480 @@
+
+        Notes
+        -----
+        Structured units can be written to all formats, but can be
+        re-read only with 'generic'.
+
+        """
+        parts = [part.to_string(format) for part in self.values()]
+        out_fmt = '({})' if len(self) > 1 else '({},)'
+        if format == 'latex':
+            # Strip $ from parts and add them on the outside.
+            parts = [part[1:-1] for part in parts]
+            out_fmt = '$' + out_fmt + '$'
+        return out_fmt.format(', '.join(parts))
+
+    def _repr_latex_(self):
+        return self.to_string('latex')
+
+    __array_ufunc__ = None
+
+    def __mul__(self, other):
+        if isinstance(other, str):
+            try:
+                other = Unit(other, parse_strict='silent')
+            except Exception:
+                return NotImplemented
+        if isinstance(other, UnitBase):
+            new_units = tuple(part * other for part in self.values())
+            return self.__class__(new_units, names=self)
+        if isinstance(other, StructuredUnit):
+            return NotImplemented
+
+        # Anything not like a unit, try initialising as a structured quantity.
+        try:
+            from .quantity import Quantity
+            return Quantity(other, unit=self)
+        except Exception:
+            return NotImplemented
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        if isinstance(other, str):
+            try:
+                other = Unit(other, parse_strict='silent')
+            except Exception:
+                return NotImplemented
+
+        if isinstance(other, UnitBase):
+            new_units = tuple(part / other for part in self.values())
+            return self.__class__(new_units, names=self)
+        return NotImplemented
+
+    def __rlshift__(self, m):
+        try:
+            from .quantity import Quantity
+            return Quantity(m, self, copy=False, subok=True)
+        except Exception:
+            return NotImplemented
+
+    def __str__(self):
+        return self.to_string()
+
+    def __repr__(self):
+        return f'Unit("{self.to_string()}")'
+
+    def __eq__(self, other):
+        try:
+            other = StructuredUnit(other)
+        except Exception:
+            return NotImplemented
+
+        return self.values() == other.values()
+
+    def __ne__(self, other):
+        if not isinstance(other, type(self)):
+            try:
+                other = StructuredUnit(other)
+            except Exception:
+                return NotImplemented
+
+        return self.values() != other.values()
+
+
+class Structure(np.void):
+    """Single element structure for physical type IDs, etc.
+
+    Behaves like a `~numpy.void` and thus mostly like a tuple which can also
+    be indexed with field names, but overrides ``__eq__`` and ``__ne__`` to
+    compare only the contents, not the field names are ignored.  Furthermore,
```

Comment:
```suggestion
    compare only the contents, not the field names.  Furthermore,
```

Reply:
Hmm, and then I had another duplication of stuff in the comment. So, fixed both.